backend/danswer/db/search_settings.py

```diff
@@ -183,7 +183,7 @@ def update_current_search_settings(
 
     # Whenever we update the current search settings, we should ensure that the local reranking model is warmed up.
     if (
-        current_settings.provider_type is None
+        search_settings.rerank_provider_type is None
```

Comment:
For my own edification, why does this check for rerank_provider_type?

Reply:
Our most generalizable method of checking for cloud reranking models (which don't require a warm up).